### ISSUE TITLE
Problems with class path

### ISSUE THREAD

**Issue by ChrisHird** [IBM/nagios-for-i#17]
When running the host_config.sh script it errors 

 /bin/bash /usr/local/nagios/libexec/host_config.sh -i host
Input the host address(IP):sas3.shield.local
Input the userID:usr
Input the password:
Confirm the password:
Exception in thread "main" java.lang.NoClassDefFoundError: com/ibm/as400/access/AS400
        at com.ibm.nagios.config.HostConfig.Insert(HostConfig.java:113)
        at com.ibm.nagios.config.HostConfig.main(HostConfig.java:82)
Caused by: java.lang.ClassNotFoundException: com.ibm.as400.access.AS400
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 2 more

We had to update the script to add the jt400.jar to the class path for it to work. I think that is because you override the class path with the -cp option?
#!/bin/bash
java -cp /usr/local/nagios/libexec/jt400.jar:/usr/local/nagios/libexec/nagios4i.jar com.ibm.nagios.config.HostConfig $1 $2

**Comment by jbeaujour** [comment]
Hello Chris
I had the same pb, how did you resolve it ?
Thank's

**Comment by ChrisHird** [comment]
We altered the classpath in the script, but we found that using the wizards worked OK in the end for Nagios. We detailed our install in a pdf file embedded in the following Blog posting.
https://www.shieldadvanced.com/Blog/uncategorized/nagios-monitoring/

We have now built our own interface and install because we could not get the responses from IBM (the support is pretty thin) for a number of issues we came across. This based on Nagios but runs on a RPi, it is developed by another developer but we added the IBM i monitoring (the next release 1.6 will have it included as we sent the developer the scripts to install) and a lot of other monitors we have developed for our HA product and items not covered by the IBM offering.

If you are interested here is a link to the web page for the product : https://nemslinux.com/  Here is a Blog posting about how we set it all up just for IBM i : https://www.shieldadvanced.com/Blog/application-modernization/nems-and-ibm-i-monitoring/

We are announcing some new features with NEMS and message monitoring on IBM i in the coming weeks.

Hope the information helps.

Chris…
